### ISSUE TITLE
external idmap does not work

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#2179]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/2175
- Created at 2013-12-06 17:25:35 by [aigoshin](https://pagure.io/user/aigoshin)
- Closed as Invalid
- Assigned to nobody
---

i configured sssd to work in "id_provider = ldap" mode for several
domains. it works as expected.

i need to make idmap sid<->uid/gid using corresponding attributes
of this ldap (ldap_id_mapping = False).

i add sid for a user, sid comes from ldap server with other attributes.
getent passwd stops returning anything, in logs i see errors.

is it my misconfiguration or a bug?

log and sssd.conf are attached.

### Comments

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2013-12-06 17:26:16](https://pagure.io/SSSD/sssd/issue/2175#239770)**

attachment
[log](/SSSD/sssd/issue/raw/files/5c3b0afa6bb22994a9bf4a1605aeb4356cbc68b1af9b871054068d0fb585e2fa-log)

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2013-12-06 17:27:20](https://pagure.io/SSSD/sssd/issue/2175#239771)**

attachment
[sssd.conf](/SSSD/sssd/issue/raw/files/2db53fc6142d1ae202a18aa1ca077003213a4c51fde11567aa8b0765402c07fa-sssd.conf)

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2013-12-09 09:41:57](https://pagure.io/SSSD/sssd/issue/2175#239772)**

This might be a duplicate of #2172

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2013-12-12 13:41:54](https://pagure.io/SSSD/sssd/issue/2175#239773)**

Actually this is not a duplicate of #2172, but seems related. I suspect that the root cause might be similar.

Does it help you if you specify your domain SID manually using ldap_idmap_default_domain_sid in the config file?

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2013-12-12 13:43:03](https://pagure.io/SSSD/sssd/issue/2175#239774)**

Fields changed

summary: exterlal idmap does not work => external idmap does not work

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2013-12-12 15:18:05](https://pagure.io/SSSD/sssd/issue/2175#239775)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.11.4

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2013-12-12 15:21:59](https://pagure.io/SSSD/sssd/issue/2175#239776)**

Linked to Bugzilla bug: https://bugzilla.redhat.com/show_bug.cgi?id=1037653 (*Red Hat Enterprise Linux 7*)

rhbz:  => [https://bugzilla.redhat.com/show_bug.cgi?id=1037653 1037653]

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2013-12-12 15:40:48](https://pagure.io/SSSD/sssd/issue/2175#239777)**

if i specify ldap_idmap_default_domain_sid sssd does not start. i tried it before opening this bug.

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2013-12-12 18:13:49](https://pagure.io/SSSD/sssd/issue/2175#239778)**

Replying to [comment:6 aigoshin]:
> if i specify ldap_idmap_default_domain_sid sssd does not start. i tried it before opening this bug.

Can you also paste the error you get in the logs trying to specify the ldap_idmap_default_domain_sid?

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2013-12-12 18:30:19](https://pagure.io/SSSD/sssd/issue/2175#239779)**

attachment
[sssd.log](/SSSD/sssd/issue/raw/files/e8a53a3b6ec1cf454430f1a222e6495292404c1614208bfb1d2d50c33ec25fe9-sssd.log)

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2013-12-12 18:30:30](https://pagure.io/SSSD/sssd/issue/2175#239780)**

attachment
[sssd_econ.vsu.ru.log](/SSSD/sssd/issue/raw/files/7a137919258efc4369d3a6a20a6c401871a63156d72a5b8f9ac882a49ab0ae04-sssd_econ.vsu.ru.log)

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2013-12-12 18:30:40](https://pagure.io/SSSD/sssd/issue/2175#239781)**

attachment
[sssd_nss.log](/SSSD/sssd/issue/raw/files/4f8f6249d1484b4d159e0951a046da8e9eee69462e3bacf6051f31859423cf31-sssd_nss.log)

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2013-12-12 18:30:49](https://pagure.io/SSSD/sssd/issue/2175#239782)**

attachment
[sssd_pam.log](/SSSD/sssd/issue/raw/files/2d1464b48ea4c3c91561919bbe7f875fb593585888e741af0a11fcbd4ba6a464-sssd_pam.log)

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2013-12-12 18:30:59](https://pagure.io/SSSD/sssd/issue/2175#239783)**

attachment
[sssd_rgph.vsu.ru.log](/SSSD/sssd/issue/raw/files/850a5bb33b4f1bdbeeaba83c74e4ae0d4bf4f0843fa24bf2e5faee7bb779242f-sssd_rgph.vsu.ru.log)

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2013-12-12 18:31:08](https://pagure.io/SSSD/sssd/issue/2175#239784)**

attachment
[sssd_vsu.ru.log](/SSSD/sssd/issue/raw/files/0c9ab9dbab867f19be2c85332119a9c7087f46afdab2d014d12dd788cd9d0791-sssd_vsu.ru.log)

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2013-12-12 18:31:49](https://pagure.io/SSSD/sssd/issue/2175#239785)**

attached.

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2014-01-22 18:04:25](https://pagure.io/SSSD/sssd/issue/2175#239786)**

Replying to [comment:8 aigoshin]:

This problem looks like a https://fedorahosted.org/sssd/ticket/2172. Patches were pushed to upstream repo for that ticket. Does it solve your problem? Do you need a help with building rpms from git repo?

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2014-01-22 18:35:51](https://pagure.io/SSSD/sssd/issue/2175#239787)**

Fields changed

cc:  => lslebodn@redhat.com

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2014-01-29 20:24:46](https://pagure.io/SSSD/sssd/issue/2175#239788)**

Because we believe this problem was solved already with another commit and at this point we're just waiting on a confirmation from the reporter, I'm moving the ticket to 1.11.5 in order to not block the 1.11.4 release.

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2014-01-29 20:25:41](https://pagure.io/SSSD/sssd/issue/2175#239789)**

Fields changed

milestone: SSSD 1.11.4 => SSSD 1.11.5

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2014-02-06 17:16:30](https://pagure.io/SSSD/sssd/issue/2175#239790)**

aigoshin, could you test sssd from git branch sssd-1-11?

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2014-02-24 19:56:43](https://pagure.io/SSSD/sssd/issue/2175#239791)**

There was no response for 3 weeks. We believe the issue is fixed in 1.11.4. Please reopen if you are able to reproduce with 1.11.4 or later.

resolution:  => worksforme
status: new => closed

---

**Comment from [aigoshin](https://pagure.io/user/aigoshin) at [2017-02-24 15:00:54](https://pagure.io/SSSD/sssd/issue/2175#239792)**

**Metadata Update from @aigoshin**:
- Issue set to the milestone: SSSD 1.11.5
